Sources/Event.swift

```diff
@@ -848,14 +850,43 @@ extension Signal.Event {
 			return { event in
 				switch event {
 				case let .value(value):
+					state.modify { state in
+						state.pendingValue = value
+					}
 					let date = scheduler.currentDate.addingTimeInterval(interval)
 					d.inner = scheduler.schedule(after: date) {
-						action(.value(value))
+						state.modify { state in
+							defer {
+								if state.pendingValue != nil {
+									state.pendingValue = nil
+									state.previousDate = date
+								}
```

Comment:
I think this logic (repeated in `throttle`) should be extracted into a method on `ThrottleState`.